src/agoradatatools/etl/transform/rna_de_aggregate.py

```diff
@@ -0,0 +1,188 @@
+        "gene_symbol"
+    ].to_dict()
+    model_info_dict = model_info_df.set_index("model")["model_type"].to_dict()
+
+    # Create label map dictionaries for efficient lookups
+    label_map_dict = {}
+    model_group_dict = {}
+    for _, row in rnaseq_genotype_label_map_df.iterrows():
+        key = (row["model"], row["genotype"])
+        label_map_dict[key] = row["display_label"]
+        model_group_dict[row["model"]] = row["model_group"]
+
+    # Create biodomain lookup dictionary
+    biodomain_dict = (
+        biodom_genes_mm_df.groupby("ensembl_id")["biodomain"].apply(list).to_dict()
+    )
+
+    output = []
+
+    # Process files one at a time to reduce memory usage
+    file_list = datasets["rna_de_aggregate_data_files"]
+    total_files = len(file_list)
+    syn = _login_to_synapse()
+
+    for i, (file_name, syn_id) in enumerate(file_list.itertuples(index=False)):
+        # Download and process one file at a time
+        data_file = get_entity_as_df(syn_id=syn_id, source="csv", syn=syn)
+        logger.info(
+            f"Processing {file_name} ({i+1}/{total_files}): {len(data_file)} rows, {len(data_file.columns)} columns, {data_file.memory_usage(deep=True).sum() / 1024**2:.2f} MB"
+        )
+
+        _quick_validate_data_file(file_name, data_file)
+
+        # Filter out rows with human gene ensembl IDs (ENSG*), keep only mouse (ENSMUSG*)
+        data_file = data_file[data_file["ensembl_gene_id"].str.startswith("ENSMUSG")]
+
+        # Group by gene, model, tissue, and sex to create one entry per group
+        grouped = data_file.groupby(["ensembl_gene_id", "model", "tissue", "sex"])
+
+        for i, ((ensembl_gene_id, model, tissue, sex), group) in enumerate(grouped):
+            # Get gene metadata using dictionary lookup
+            gene_symbol = gene_metadata_dict.get(ensembl_gene_id, "")
+
+            # Get case and control from first row of group
+            case = group.iloc[0]["case"]
+            control = group.iloc[0]["control"]
+
+            # Use dictionary lookups instead of .loc[] operations
+            name = label_map_dict.get((model, case), model)
+            matched_control = label_map_dict.get((model, control), model)
+            model_group = model_group_dict.get(model)
+
+            # Get biodomains using dictionary lookup
+            biodomains = biodomain_dict.get(ensembl_gene_id, [])
+
+            # Get model type using dictionary lookup
+            model_type = model_info_dict.get(model, "")
+
+            # Create age-based entries
+            age_entries = {}
+            for _, row in group.iterrows():
+                age = str(row["age"])
+                age_entries[age] = {
+                    "log2_fc": float(f"{float(row['log2FoldChange']):.5g}"),
+                    "adj_p_val": float(f"{float(row['padj']):.5g}"),
+                }
+
+            # If "name" includes "Jax" (case-insensitive) and tissue is "Right Cerebral Hemisphere", change tissue to "Hemibrain"
```

Comment:
Make sonarcloud happy:
```suggestion
            # If "name" includes "Jax" (case-insensitive) and tissue is "Right Cerebral Hemisphere",
            # change tissue to "Hemibrain"
```

Reply:
Thank you!